clang/test/CIR/CodeGen/loop.cpp

```diff
@@ -190,6 +190,117 @@ void l3() {
 // OGCG:   store i32 0, ptr %[[I]], align 4
 // OGCG:   br label %[[FOR_COND]]
 
+void l4() {
+  int a[10];
+  for (int n : a)
```

Comment:
Can we do a test with a type with `begin` and `end` as well?  Or cant we support general iterators yet?